bundle-workflow/src/test_workflow/dependency_installer.py

```diff
@@ -0,0 +1,62 @@
+import os
+import shutil
+
+from s3_utility import S3ReadWrite
+
+
+class DependencyInstaller:
+    """
+    Provides a dependency installer for the test suites.
+    """
+
+    def __init__(self, build):
+        self.build_id = build.id
+        self.version = build.version
+        self.arch = build.architecture
+        self.s3_utility = S3ReadWrite()
+        self.s3_maven_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/maven"
+        )
+        self.s3_build_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/plugins"
+        )
+
+    def installMavenDependencies(self, dependency_list):
+        """
+        Downloads the maven dependencies from S3 and puts them on the maven local path
+        for each dependency in the dependency_list.
+
+        :param dependency_list: list of dependencies for which the maven artifacts need to be downloaded.
+        """
+        for dependency in dependency_list:
+            s3_path = f"{self.s3_maven_location}/{dependency}"
+            maven_local_path = os.path.join(
+                os.path.expanduser("~"),
+                f".m2/repository/org/opensearch/{dependency}/{self.version}/",
+            )
+            self.s3_utility.download_folder(s3_path, maven_local_path)
+
+    def installBuildDependencies(self, dependency_list, custom_local_path):
+        """
+        Downloads the build dependencies from S3 and puts them on the given custom path
+        for each dependency in the dependency_list.
+
+        :param dependency_list: list of dependencies for which the build artifacts need to be downloaded.
```

Comment:
it should be list of dependency names

Reply:
need to change the code to handle this

Reply:
I will edit the comment to dependency names.